### PR TITLE
Fixed high token usage for json type

### PR DESCRIPTION
**Pull request by Deepak-Kesavan**
## What

- Fixed high token usage for json type

## Why

- To stop unwanted token usage

## How

- Try parsing the LLM response to json before performing another LLM call to give answer in json format.

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)

- No. It is just enhancement

## Database Migrations

- 

## Env Config

- 

## Relevant Docs

-

## Related Issues or PRs

-

## Dependencies Versions

-

## Notes on Testing

-

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: prompt-service/src/unstract/prompt_service/main.py [hunk 1]
```diff
@@ -497,24 +497,29 @@ def prompt_processor() -> Any:
...
+                            )
+                            structured_output[output[PSKeys.NAME]] = json.loads(answer)
+                        except JSONDecodeError as e:
+                            app.logger.info(
+                                f"JSON format error : {answer}", LogLevel.ERROR
+                            )
+                            app.logger.info(
+                                f"Error parsing response (to json): {e}", LogLevel.ERROR
+                            )
+                            structured_output[output[PSKeys.NAME]] = {}
```
Comment (chandrasekharan-zipstack) [inline prompt-service/src/unstract/prompt_service/main.py:521]: NIT: Make a function if possible to reduce nesting and cyclomatic complexity

**Review by chandrasekharan-zipstack — APPROVED** [review]

File: prompt-service/src/unstract/prompt_service/main.py [hunk 2]
```diff
@@ -546,6 +551,7 @@ def prompt_processor() -> Any:
                             usage_kwargs=usage_kwargs,
                         )
                         challenge = challenge_plugin["entrypoint_cls"](
+                            llm=llm,
```
Comment (gaya3-zipstack) [inline prompt-service/src/unstract/prompt_service/main.py:554]: Do we have a corresponding PR to handle this new param for challenge @Deepak-Kesavan ?

Comment (gaya3-zipstack) [reply prompt-service/src/unstract/prompt_service/main.py:554]: Got it - https://github.com/Zipstack/unstract-cloud/pull/119

**Review by gaya3-zipstack — APPROVED** [review]
Looks good.

**Pull request merged**